Index: components/ideintf/objectinspector.lfm
===================================================================
--- components/ideintf/objectinspector.lfm	(revision 47342)
+++ components/ideintf/objectinspector.lfm	(working copy)
@@ -1,7 +1,7 @@
 object ObjectInspectorDlg: TObjectInspectorDlg
-  Left = 338
+  Left = 483
   Height = 669
-  Top = 162
+  Top = 103
   Width = 300
   BorderStyle = bsSizeToolWin
   Caption = 'ObjectInspectorDlg'
@@ -8,11 +8,11 @@
   ClientHeight = 669
   ClientWidth = 300
   KeyPreview = True
-  LCLVersion = '1.3'
+  LCLVersion = '1.4.0.1'
   object StatusBar: TStatusBar
     Left = 0
-    Height = 23
-    Top = 646
+    Height = 21
+    Top = 648
     Width = 300
     Panels = <    
       item
@@ -24,11 +24,11 @@
   end
   object AvailPersistentComboBox: TComboBox
     Left = 0
-    Height = 23
+    Height = 31
     Top = 0
     Width = 300
     Align = alTop
-    ItemHeight = 15
+    ItemHeight = 0
     OnCloseUp = AvailComboBoxCloseUp
     Style = csDropDownList
     TabOrder = 0
@@ -36,9 +36,10 @@
   object ComponentPanel: TPanel
     Left = 0
     Height = 184
-    Top = 23
+    Top = 31
     Width = 300
     Align = alTop
+    BevelOuter = bvNone
     ClientHeight = 184
     ClientWidth = 300
     TabOrder = 2
@@ -47,10 +48,10 @@
       AnchorSideLeft.Side = asrBottom
       AnchorSideTop.Control = FilterLabel
       AnchorSideTop.Side = asrCenter
-      Left = 92
-      Height = 28
-      Top = 3
-      Width = 204
+      Left = 94
+      Height = 27
+      Top = 2
+      Width = 202
       UseFormActivate = True
       ButtonWidth = 23
       NumGlyphs = 1
@@ -61,9 +62,9 @@
     end
     object FilterLabel: TLabel
       Left = 3
-      Height = 15
+      Height = 17
       Top = 7
-      Width = 69
+      Width = 86
       Caption = 'Co&mponents'
       FocusControl = CompFilterEdit
       ParentColor = False
Index: ide/include/unix/lazbaseconf.inc
===================================================================
--- ide/include/unix/lazbaseconf.inc	(revision 47342)
+++ ide/include/unix/lazbaseconf.inc	(working copy)
@@ -142,12 +142,16 @@
   procedure InternalInit;
  ---------------------------------------------------------------------------}
 procedure InternalInit;
+var
+  S: string;
 begin
   // For the Unix file functions only the slash is a directory separator.
   // The RTL defines AllowDirectorySeparators ['/','\'] for historical reasons.
   AllowDirectorySeparators:=['/'];
-
-  PrimaryConfigPath:=ExpandFileNameUTF8('~/.lazarus');
+  S := ExpandFileNameUTF8(ParamStrUTF8(0));
+  S := ExtractFileDir(S);
+  S := IncludeTrailingPathDelimiter(S) + 'config';
+  PrimaryConfigPath:= S;
   SecondaryConfigPath:='/etc/lazarus';
 end;
 
Index: ide/main.pp
===================================================================
--- ide/main.pp	(revision 47342)
+++ ide/main.pp	(working copy)
@@ -2031,17 +2031,19 @@
     Result.Parent := MainIDEBar.pnlSpeedButtons;
     Result.Images := IDEImages.Images_16;
     Result.AutoSize := true;
+    Result.EdgeBorders := [];
   end;
 
 begin
   // Panel for buttons on the left
   MainIDEBar.pnlSpeedButtons := TPanel.Create(OwningComponent);
-  with MainIDEBar.pnlSpeedButtons do 
+  with MainIDEBar.pnlSpeedButtons do
   begin
     Name := 'pnlSpeedButtons';
     Parent := MainIDEBar;
     Align := alLeft;
     Caption := '';
+    BorderSpacing.Top := 8;
     BevelOuter := bvNone;
     AutoSize := true;
     Visible := EnvironmentOptions.IDESpeedButtonsVisible;
Index: ide/projectinspector.lfm
===================================================================
--- ide/projectinspector.lfm	(revision 47342)
+++ ide/projectinspector.lfm	(working copy)
@@ -9,16 +9,15 @@
   ClientHeight = 456
   ClientWidth = 299
   OnDropFiles = FormDropFiles
-  LCLVersion = '1.3'
+  LCLVersion = '1.4.0.1'
   object ItemsTreeView: TTreeView
     Left = 0
-    Height = 378
-    Top = 78
+    Height = 403
+    Top = 53
     Width = 299
     Align = alClient
     Anchors = [akTop, akLeft, akBottom]
     BorderSpacing.Top = 2
-    DefaultItemHeight = 18
     DragMode = dmAutomatic
     MultiSelect = True
     PopupMenu = ItemsPopupMenu
@@ -36,7 +35,7 @@
   object BtnPanel: TPanel
     Left = 0
     Height = 28
-    Top = 48
+    Top = 23
     Width = 299
     Align = alTop
     ClientHeight = 28
@@ -103,8 +102,8 @@
       AnchorSideTop.Control = SortAlphabeticallyButton
       AnchorSideTop.Side = asrCenter
       Left = 91
-      Height = 28
-      Top = 0
+      Height = 27
+      Top = 1
       Width = 205
       UseFormActivate = True
       ButtonWidth = 23
@@ -120,12 +119,14 @@
   end
   object ToolBar: TToolBar
     Left = 0
-    Height = 48
+    Height = 23
     Top = 0
     Width = 299
     AutoSize = True
-    ButtonHeight = 46
-    ShowCaptions = True
+    ButtonHeight = 23
+    EdgeBorders = []
+    EdgeInner = esNone
+    EdgeOuter = esNone
     TabOrder = 2
   end
   object ItemsPopupMenu: TPopupMenu
